models/users.ts

```diff
@@ -0,0 +1,69 @@
+import { InferRawDocType, Schema, model } from 'mongoose';
+import { TypeWithId } from '../utils/types';
+
+const userSchemaDefinition = {
+	username: {
+		type: String,
+		required: true,
+		unique: true,
+	},
+	email: {
+		type: String,
+		required: true,
+		unique: true,
+	},
+	password: {
+		type: String,
+		required: true,
+	},
+} as const;
+
+const userSchema = new Schema(userSchemaDefinition);
+
+export const userModel = model('users', userSchema);
+
+export type User = TypeWithId<InferRawDocType<typeof userSchemaDefinition>>;
+
+export type UserDetails = Pick<User, 'password' | 'email'>;
```

Comment:
Did u mean email and username?
If so u can just `Omit` password

Reply:
I did mean these because I don't know what other fields we might add and I don't want them to be automatically added to the filters